app/controllers/api/v1/users_controller.rb

```diff
@@ -110,14 +114,15 @@ def index
     #{json_schema(Api::V1::UserRepresenter, include: :readable)}
   EOS
   def show
+    return head(:forbidden) if current_human_user.is_anonymous?
     ScoutHelper.ignore!(0.999)
 
     OSU::AccessPolicy.require_action_allowed!(:read, current_api_user, current_human_user)
 
     SetGdprData.call(user: current_human_user,
                      headers: request.headers,
                      session: session,
-                     ip: ENV['IP_ADDRESS_FOR_GDPR'] || request.ip)
+                     ip: ENV.fetch('IP_ADDRESS_FOR_GDPR', nil) || request.ip)
```

Comment:
Since fetch() has a default nil value, would request,ip ever be used? Might be some Rails magic I don't know about. The [docs](https://ruby-doc.org/core-3.1.2/ENV.html#method-c-fetch) for fetch()